test/topology_random_failures/test_random_failures.py

```diff
@@ -149,7 +147,7 @@ async def test_random_failures(manager: ManagerClient,
 
     server_log = await manager.server_open_log(server_id=s_info.server_id)
 
-    if cluster_event_duration + 1 >= WAIT_FOR_IP_TIMEOUT and error_injection in ERROR_INJECTIONS_NODE_MAY_HANG:
+    if error_injection in ERROR_INJECTIONS_NODE_MAY_HANG:
```

Comment:
What will happen if we execute this `if` while the node is still in the process of joining the cluster, but the topology coordinator will later decide that joining takes too long and will reject the node?

Reply:
The test will fail to wait for "Startup failed" or "initialization completed" messages below.